.github/actions/build-fedora/action.yaml

```diff
@@ -0,0 +1,58 @@
+name: Build Fedora single-config
+
+inputs:
+  extra-cmake-flags:
+    required: false
+  compiler:
+    required: true
+  compiler-version:
+    required: false
+  build-type:
+    required: false
+    default: "RELWITHDEBINFO"
```

Comment:
This is likely useful, but would violate DRY since they are also in build-types/action. I'll go ahead and add it so we can get error checking from other repos (e.g., external-test). That does remind me that I need to add MINSIZEREL over there.

Reply:
Several compiler/C++ version combinations produce stack frame size warnings for MINSIZEREL, so testing MINSIZEREL can wait until that is cleaned up; it's low priority since MINSIZEREL is unlikely to be used.

Reply:
Ah, ok. I haven't tried that build type before. I'll leave it out here until we get those warnings fixed.